resources/assets/lib/utils/lang.tsx

```diff
@@ -2,12 +2,32 @@
 // See the LICENCE file in the repository root for full licence text.
 
 import Lang from 'lang.js';
+import * as React from 'react';
 import { formatNumber } from 'utils/html';
 import { present } from 'utils/string';
 
 type Replacement = string | number;
 export type Replacements = Partial<Record<string, Replacement>>;
 
+export function joinComponents(array: React.ReactNode[], key = 'common.array_and') {
+  const nodes: React.ReactFragment[] = [];
+
+  if (array.length > 0) {
+    nodes.push(<React.Fragment key={0}>{array[0]}</React.Fragment>);
+
+    const lastIndex = array.length - 1;
+    const lastConnector = lastIndex === 1 ? trans(`${key}.two_words_connector`) : trans(`${key}.last_word_connector`);
+
+    for (let i = 1; i < lastIndex; i++) {
+      nodes.push(<React.Fragment key={i}>{trans(`${key}.words_connector`)}{array[i]}</React.Fragment>);
```

Comment:
shouldn't really matter much but the words connector can be stored in a variable before the loop

Reply:
I think getting rid of the `key` param altogether would be better since we don't use it at all 👀

Reply:
but that's not related to my comment 🤔 (`const wordsConnector = trans(...)` instead of calling the same function and parameter repeatedly)